ivy_models/transformers/perceiver_io.py

```diff
@@ -46,7 +46,29 @@ def __init__(self,
         if learn_query and not ivy.exists(query_shape):
             raise Exception('if learn_query is set, then query_shape must be specified.')
 
-        device = ivy.default(device, ivy.default_device())
+        self.device = ivy.default(device, ivy.default_device())
```

Comment:
Why are all of these attributes needed now, while they were apparently not needed before?

Reply:
yes these are not required with docker, didn't realize that most of the local testing changes were not required when using docker. I have removed them.